### PR TITLE
✨(docker) create Dockerfile to build jibri with pulseaudio support

### PR DESCRIPTION
**Pull request by madmatah**
## Purpose

[Jibri](https://github.com/jitsi/jibri) provides services for recording or streaming a [Jitsi Meet](https://github.com/jitsi/jitsi-meet) conference.

It works by launching a Chrome instance rendered in a virtual framebuffer and capturing and encoding the output with ffmpeg. The audio is captured with a virtual audio device, which is a Alsa loopback device.

To use Jibri within a container (with the [official Jibri docker image](https://github.com/jitsi/docker-jitsi-meet/)), it requires to create a loopback device on the host and share it to the container. 

This approach has the following drawbacks:
 - It is not practical to run multiple instances of Jibri on the same host.
 - Some cloud based container platforms does not allow customization of hosts and therefore do not allow to create the required Alsa loopback sound device.
 - It is hard to build a dynamic pool of Jibri containers with auto-scaling features provided by some orchestrators (like [Kubernetes HPA](https://kubernetes.io/docs/tasks/run-application/horizontal-pod-autoscale/)).

We want to use pulseaudio instead of alsa in jibri docker image. 

## Proposal

This image is based on the official jibri docker image, that relies on [S6 overlay](https://github.com/just-containers/s6-overlay).
You can find the base image definition [here](https://github.com/jitsi/docker-jitsi-meet/tree/master/jibri).

The following changes have been made to the base image:

- A pulse service has been added to start pulseaudio as the jibri user.
- The jibri user has realtime scheduling capabilities in pulseaudio with rtkit
- The jibri configuration file has been updated to use the new Typesafe config file format instead of the [deprecated](https://github.com/jitsi/jibri/blob/master/doc/config_migration_guide.md) JSON format.
- `JIBRI_FFMPEG_*` environment variables have been introduced to customize ffmpeg arguments used by Jibri
- The xorg configuration is now generated to have a display resolution that matches the desired recording resolution.

To facilitate the use of this project, a docker-compose configuration is provided to run an entire Jitsi stack. A Makefile is also provided to run the common tasks, and a README with basic instructions.

### CODE REVIEW

**Comment by sampaccoud** [comment]
:astonished:

File: README.md [hunk 1]
```diff
@@ -1,2 +1,69 @@
...
+host.
+
+## Getting Started
+
+The following instructions are meant to compile and run the Jibri docker image on your
+workstation, with a fully functionning jitsi-meet stack.
+
+### Requirements
+
+  - Add `meet.jitsi` to your `/etc/hosts`
```
Comment (madmatah) [inline README.md]:
```suggestion
  - Add the following entry to your `/etc/hosts`: `127.0.0.1	meet.jitsi`
```

File: docker/files/etc/cont-init.d/10-config [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+OLD_JIBRI_XMPP_PASSWORD=passw0rd
+if [[ "$JIBRI_XMPP_PASSWORD" == "$OLD_JIBRI_XMPP_PASSWORD" ]]; then
+    echo 'FATAL ERROR: Jibri auth password must be changed, check the README'
+    exit 1
+fi
+
+# DISPLAY is necessary for start
+[ -z "${DISPLAY}" ] \
+&& ( echo -e "\e[31mERROR: Please set DISPLAY variable.\e[39m"; kill 1; exit 1 )
```
Comment (lunika) [inline docker/files/etc/cont-init.d/10-config]: In previous check there is no `kill 1` nor color in the message. We could probably homogenize this ?

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,12 @@
+ARG BASE_IMAGE=jitsi/jibri
+ARG BASE_TAG=latest
+FROM ${BASE_IMAGE}:${BASE_TAG}
+
+RUN \
+	apt-get update \
```
Comment (lunika) [inline Dockerfile]: In the base image they don't directly call `apt-get`, they call `apt-dpkg-wrap` before. `apt-dpkg-wrap apt-get update`

File: README.md [hunk 4]
```diff
@@ -24,6 +24,40 @@ The purpose of this project is to replace Alsa with Pulseaudio in Jibri, to be
...
+
+**Initialize the project**
+ - Clone the git repository
+ - `make bootstrap`
+
+**Launch the services**
+ - `make run`
+
+**Start a meeting and record it**
+ - Open [https://jitsi.meet/](https://jitsi.meet/) on your browser
```
Comment (lunika) [inline README.md]:
```suggestion
Open [https://meet.jitsi:8443/](https://meet.jitsi:8443/) on your browser and accept the https certificate if needed
```

File: README.md [hunk 5]
```diff
@@ -1,2 +1,69 @@
...
+**Launch the services**
+ - `make run`
+
+**Start a meeting and record it**
+ - Open [https://meet.jitsi/](https://meet.jitsi/) on your browser
+ - Start a recording with the jitsi-meet interface
+
+You can watch `Jibri`'s logs with `make logs`.
+
+When the recording is stopped, you'll find your recordings in `~/.jitsi-meet-cfg/jibri/recordings/`.
```
Comment (lunika) [inline README.md:56]: You can also add in the readme that this project will create a directory outside of the project itself ?

**Comment by madmatah** [comment]
@lunika Thanks for your review! I addressed all your feedbacks.

**Review by lunika — APPROVED** [review]

**Pull request merged**